src/components/molecules/SelectInput/Select/Select.tsx

```diff
@@ -0,0 +1,120 @@
+import React, { MouseEvent, useEffect, useState } from 'react';
+import { ArrowIcon } from 'assets';
+import {
+  StyledArrowImage,
+  StyledInput,
+  StyledSelect,
+  StyledSelectCaption,
+  StyledTopWrapper,
+  StyledWrapper,
+} from './Select.styled';
+import Option from '../Option';
+import { ISelect } from './Select.model';
+import { ISingleOption } from '../SelectInput.model';
+
+const Select = ({
+  isMulti,
+  options,
+  handleOnClick,
+  selectCaption,
+  inputPlaceholder,
+}: ISelect): JSX.Element => {
+  const [currentOptionId, setCurrentOptionId] = useState<string>('');
+  const [isSelectOpened, setIsSelectOpened] = useState<boolean>(false);
+  const [selectedOptions, setSelectedOptions] = useState<ISingleOption[]>([]);
+  const [mutableOptions, setMutableOptions] = useState<ISingleOption[]>(options);
+
+  const doesArrayConsistValue = (id: string) => {
+    return selectedOptions.filter((element: ISingleOption) => element.id === id).length;
+  };
+
+  const isInputEmpty = (value: string) => {
+    return value === '';
+  };
+
+  const isInputTextMatch = (inputText: string, value: string) => {
+    const regex = new RegExp(`^${inputText}`, 'i');
+    return regex.test(value);
+  };
+
+  const updateSelectedOptions = (option: ISingleOption): void => {
+    const { id } = option;
+    setCurrentOptionId(id);
+    if (isMulti) {
+      if (!doesArrayConsistValue(id)) {
+        setSelectedOptions([option, ...selectedOptions]);
+      } else {
+        const newArray = selectedOptions.filter((element: ISingleOption) => element.id !== id);
+        setSelectedOptions([...newArray]);
+      }
+    } else {
+      setSelectedOptions([option]);
+    }
+  };
+
+  const hideList = () => {
+    setIsSelectOpened(false);
+  };
+
+  const filterOptions = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = e.target.value;
+    const tempOptions: ISingleOption[] = [];
+
+    options.forEach((option) => {
```

Comment:
I am enemy of forEach :D Can we better use filter to move these results in line 70?